### ISSUE TITLE
Feature request: Support the Scala Editor

### ISSUE THREAD

**Issue by hrj** [vrapper/vrapper#629]
I see that you have special plugins for Java, Python, etc. So I wonder whether a spearate plugin for Scala is required. The Scala editor from http://scala-ide.org/ mostly works fine. But some things don't work *sometimes*. For example, folds are not opening in a file for me. Or `Ctrl-O` to open members of a class, doesn't work, sometimes.

Thx!

**Comment by hrj** [comment]
I just realized, opening folds is having problems even with mouse actions. So that is probably a Scala editor bug rather than something specific to vrapper.

**Comment by albertdev** [comment]
What are you requesting here? The reason we have Java and Python plugins is to create a number of default mappings for those languages, the editor itself should be supported out of the box.

I'll give it a try later on, but in theory we can't do anything about `Ctrl-O` failing (there *could* be a bug, but then you'd have to point to real Vrapper errors).

But again, what features would you expect in a Scala language plugin that Vrapper doesn't do yet? Please be specific as I'm not using Scala. In the Java plugin for example, we currently map `gd` and `gD` as "Go to declaration", `gr` and `gm` to open the Refactor / Source menus. We bound `=` and `gc` as operators to do indenting and toggle comment on text objects. We *could* do the same for the Scala editor if there are corresponding Eclipse commands, but then you would need to give them to me.

**Comment by hrj** [comment]
If the only thing that the language plugins do is support language specific features, then I don't have anything specific to request for Scala. I have used Vrapper mostly as a langauge-agnostic editor. (Good to learn that `gD` has special behavior in Java).

I thought the language specific plugins help overcome the quirks in that language's editor. So, I thought, a Scala specific plugin would help remove the quirks / bugs that I encounter while using vrapper with `.scala` files.

If the type of editor doesn't matter to vrapper, I can file the bugs I encounter when editing Scala code as generic vrapper bugs. Else, I will report them as Scala specific bugs in this tracker.

**Comment by keforbes** [comment]
So is there any remaining feature request or defect here?  Should I just close this issue?

For future reference, here's what our current language-specific plugins do:
http://vrapper.sourceforge.net/documentation/?topic=optional_plugins#programming
They do nothing more than define mappings to Eclipse operations which are specific to the language.

**Comment by hrj** [comment]
TLDR: There is no defect remaining. I am closing this.

-----
Thanks for the clarifications.

I found one reproducible problem with folding in Scala editor using the Vim folding commands. But I later found that it is also reproducible without vrapper, using Eclipse folding shortcuts. I will report this to the Scala editor directly. If anyone is interested, here are the details:

* Eclipse Luna + latest Scala IDE (4.0.0.v-2._11-2015)
* Create a scala project and open any `.scala` file for editing
* `zM` or `Ctrl Shift Numpad -` to fold all
* Try to open a fold with `zo` or `Ctrl Numpad +`. This fails on first attempt but works on second attempt (for every fold).
